othello/moderator/moderator.py

```diff
@@ -9,40 +11,42 @@ class InvalidMoveError(RuntimeError):
     Exception representing an invalid move given the current game state
     """
 
-    def __init__(self, board, player, move):
+    def __init__(self, board: str, player: str, move: int) -> None:
         self.code = utils.UserError.INVALID_MOVE.value[0]
         self.message = utils.UserError.INVALID_MOVE.value[1].format(
             move=move, player=constants.PLAYERS[player].value, board=utils.binary_to_string(board)
         )
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.message
 
 
 class Moderator:
     def __init__(self):
-        self.board = constants.INITIAL  # bitboards representation of the initial othello board
-        self.game_over = False
-        self.current_player = constants.BLACK  # black moves first
+        self.board: Dict[
+            int, int
+        ] = constants.INITIAL  # bitboards representation of the initial othello board
+        self.game_over: bool = False
+        self.current_player: int = constants.BLACK  # black moves first
 
-    def is_game_over(self):
+    def is_game_over(self) -> bool:
         return self.game_over
 
-    def get_board(self):
+    def get_board(self) -> str:
         """
         Returns the string representation of the current bitboards
         """
         return utils.binary_to_string(self.board)
 
-    def toggle_current_player(self):
+    def toggle_current_player(self) -> None:
         """
         Switches the current player
         1 ^ 1 = 0
         0 ^ 1 = 1
         """
         self.current_player ^= 1
 
-    def outcome(self):
+    def outcome(self) -> Union[str, bool]:
```

Comment:
In my opinion, this should be refactored to return an `Optional[str]`, with `None` indicating that the game is not over.